Mark disable-affinity-assistant as deprecated (#6991)
tektoncd/pipeline: docs/deprecations.md

| [The `default-cloud-events-sink` setting in the `config-defaults` configMap is deprecated](https://github.com/tektoncd/pipeline/pull/6883) in favour of the new `config-events` configMap.                                                                                                                | v0.50.0                                                              | N/A                                                                                                  | v0.59.0                       |
| [v1beta1 Tasks, TaskRuns, Pipelines, and PipelineRuns are deprecated in favor of v1](https://github.com/tektoncd/pipeline/issues/5541)                                                                                                                 | v0.50.0                                                              | Beta                                                                                                  | v0.62.0                       |

> | [Deprecate and remove disable-affinity-assistant feature flag](https://github.com/tektoncd/pipeline/issues/6988) in favour of the new `coschedule` configMap                                                                                                                 | v0.51.0                                                              | N/A                                                                                                  | April 27, 2024                       |

Reviewer: ```suggestion
| [The `disable-affinity-assistant` feature flag is deprecated in favor of the `coschedule` feature flag](https://github.com/tektoncd/pipeline/issues/6988)                                                                                                                | v0.51.0                                                              | N/A                                                                                                  | April 27, 2024                       |
```

Author: Changed
